spec/spec_helper.rb

```diff
@@ -277,6 +312,27 @@ def assert_faster_than(competition, records, &blk)
     upsert_time.should be < ar_time
     $stderr.puts "   Upsert was #{((ar_time - upsert_time) / ar_time * 100).round}% faster than #{competition}"
   end
+
+  def clone_ar_class(klass, table_name)
+    u = Upsert.new $conn, klass.table_name
+    new_table_name = [*table_name].compact
+    # AR's support for quoting of schema and table names is horrendous
+    # schema.table and schema.`table` are considiered different names on MySQL, but
+    # schema.table and schema."table" are correctly considered the same on Postgres
+    new_table_name[-1] = u.connection.quote_ident(new_table_name[-1]) if new_table_name[-1].to_s.index('.')
+    new_table_name = new_table_name.join('.')
+    cls = Class.new(klass)
+    cls.class_eval do
+      self.table_name = new_table_name
+      def self.quoted_table_name
+        table_name
```

Comment:
overriding `quoted_table_name` may change behavior significant in a lib that writes sql... necessary?

Reply:
This is only for the spec, and we can remove it when we take a look at removing `active_record_inline_schema` :)